### ISSUE TITLE
Replacement for QWebPage::javaScriptConsoleMessage()

### ISSUE THREAD

**Issue by Emdek** [annulen/webkit#144]
`QWebPage::javaScriptConsoleMessage()` doesn't provide enough information (for example log level, available in QtWebEngine) and is limited only to JavaScript.
Existing method could be marked as deprecated and replaced by something similar to:
`QWebEnginePage::consoleMessage(ConsoleMessageSource source, ConsoleMessageLevel level, const QString &message, int lineNumber, const QString &sourceID)`

*sourceID* is a name used by existing method but it could get renamed since WebKit uses *Source* in different context.

**Comment by andrwwbstr** [comment]
We apply a patch that does this.  It uses a variant of javaScriptConsoleMessage instead:

```patch
     virtual void javaScriptConsoleMessage(const QString& message, int lineNumber, const QString& sourceID);
+    virtual void javaScriptConsoleMessage(MessageSource source, MessageLevel level, const QString& message, int lineNumber, const QString& sourceID);
```

I believe for backwards compatibility both variants should be called.


I can share the full patch if anyone is interested, but it will need a bit of work.

**Comment by annulen** [comment]
That would be great!

**Comment by Emdek** [comment]
@andrwwbstr, sure, deprecated doesn't mean dead. ;-)
Although new shouldn't have the same name since it's wrong, misleading (it's not limited to that source anymore).

**Comment by andrwwbstr** [comment]
See #166 for the original commit.  Although some changes are probably required (e.g. name changes, or eliminating the call to each variant), it could be used as a starting point.

**Comment by annulen** [comment]
Fixed in #166
